src/uniprotkb/components/entry/GoRibbon.tsx

```diff
@@ -65,7 +65,39 @@ const GoRibbon = ({
     'protvista-datatable'
   );
 
-  const [selectedSet, setSelectedSet] = useState('goslim_generic');
+  const [selectedSet, setSelectedSet] = useState(() => {
+    let defaultSS = 'goslim_generic';
+    if (organismData?.scientificName && organismData?.lineage) {
+      const taxonomyInfo = [
+        ...organismData.lineage,
+        organismData.scientificName,
+      ];
+
+      // SlimSets based on Taxonomy
+      const slimSetByTaxon = {
+        goslim_plant: [
+          'Viridiplantae',
+          'Bangiophyceae',
+          'Florideophyceae',
+          'Stylonematophyceae',
+          'Rhodellophyceae',
+          'Compsopogonophyceae',
+        ],
+        prokaryotes: ['Bacteria', 'Archaea'],
```

Comment:
So, one thing that came out of the conversation with SIB was to use a prokaryote slimming set, but the issue is that we don't have such slimming set available 🤷🏽 
It's good that this logic is here, but it should probably be commented out for now because it renders "agr" in the select element and doesn't render any visualisation.
We'll have further discussions regarding this to see what we can and cannot do, the point for now is to have this PR's logic in place
```suggestion
        // prokaryotes: ['Bacteria', 'Archaea'],
```

Reply:
Ah, yes it takes 'agr', I didn't notice that. I will comment it for now